### ISSUE TITLE
Support Typescript 2.4.x

### ISSUE THREAD

**Issue by SethDavenport** [angular-redux/router#17]
### This is a...

* [ ] feature request
* [x] bug report
* [ ] usage question

### What toolchain are you using for transpilation/bundling?

* [ ] @angular/cli
* [ ] Custom @ngTools/webpack
* [ ] Raw `ngc`
* [ ] SystemJS
* [ ] Rollup
* [ ] Other

### Environment

NodeJS Version:
Typescript Version:
Angular Version:
@angular-redux/store version:
@angular/cli version: (if applicable)
OS:

### Link to repo showing the issus
(optional, but helps _a lot_)

### Expected Behaviour:

This should work fine (it does in typescript < 2.4)

```typescript
import { routerReducer } from '@angular-redux/router';

export const rootReducer = composeReducers(
  combineReducers({
    // ... etc.
    router: routerReducer,
  }));
```

### Actual Behaviour:

You get a TSC error (below).

### Stack Trace/Error Message:

```sh
ERROR in /Users/sdavenport/code/angular-redux/example-app/src/app/store/reducers.ts (12,19): Argument of type '{ router: routerReducer...' is not assignable to parameter of type 'ReducersMapObject'.
  Property 'router' is incompatible with index signature.
    Type '(state: string, action: RouterAction) => string' is not assignable to type 'Reducer<any>'.
      Types of parameters 'action' and 'action' are incompatible.
        Type 'A' is not assignable to type 'RouterAction'.
          Type 'Action' is not assignable to type 'RouterAction'.
            Property 'payload' is missing in type 'Action'.
```

### Additional Notes:
(optional)

**Comment by SethDavenport** [comment]
Typescript's stricter checks for higher order functions have exposed a bug in the Redux typings, discussion here: https://github.com/reactjs/redux/pull/2467

A fix has been merged to Redux but it's waiting for a new npm release. In the meantime we might be able to get away with a downcast in the implementation of `routerReducer`.

**Comment by SethDavenport** [comment]
Fixed in @angular-redux/router@6.3.1.
